sdk/rust/src/lib.rs

```diff
@@ -350,6 +444,31 @@ impl Client {
         }
     }
 
+    /// Sends an HTTP request to the server and decodes the response.
+    ///
+    /// # Developer Notes
+    ///
+    /// - Constructs an HTTP request using the specified route and optional query parameters.
+    /// - If an `api_key` is set, it adds a `Bearer` authorization header to the request.
+    /// - This function expects the server response to conform to the following structure:
+    ///    - `DecodedResponse::Ok` for successful responses.
+    ///    - `DecodedResponse::Err` for error bodies returned by the server.
+    /// - The function uses `reqwest::Client` internally and decodes the response using `serde`.
+    ///
+    /// # Parameters
+    ///
+    /// - `route` - Defines the API endpoint and HTTP method via the `Routable` trait.
+    /// - `query` - Optional query parameters that are serialized into the request URL.
+    ///
+    /// # Implementation Details
+    ///
+    /// - If the HTTP response is valid but contains an error body, the function returns a
```

Comment:
seems vague... maybe you can explain this better?

Reply:
like this happens when the client makes an invalid request,
however for some types of invalid requests like the 422 due to a negative number I guess it would go to `ClientError::DecodeResponseFailed` since they are not within `ErrorBodyResponse`